.travis.yml

```
@@ -1,55 +1,35 @@
 sudo: true
-language: haskell
+language: generic
 
 git:
   depth: 5
 
-cabal: "3.0"
-
 cache:
   directories:
-  - "$HOME/.cabal/store"
   - "$HOME/.stack"
-  - "$TRAVIS_BUILD_DIR/.stack-work"
 
 matrix:
   include:
+  - env: GHC='8.8.3'
+    os: linux
+  - env: GHC='8.8.3'
+    os: osx
 
-  # Cabal
-  - ghc: 8.2.2
-  - ghc: 8.4.4
-  - ghc: 8.6.5
-  - ghc: 8.8.2
-
-  # stack
-  - ghc: 8.8.1
-    env: STACK_YAML="$TRAVIS_BUILD_DIR/stack.yaml"
+before_install:
+  # HLint check
+  - curl -sSL https://raw.github.com/ndmitchell/neil/master/misc/travis.sh | sh -s -- hlint co-log-core/ co-log/src/ co-log-benchmark/
 
 install:
-  - |
-    if [ -z "$STACK_YAML" ]; then
-      cabal build all --enable-tests --enable-benchmarks
-    else
-      # install stack
-      curl -sSL https://get.haskellstack.org/ | sh
-
-      # build project with stack
-      stack --version
-      stack build --system-ghc --test --no-run-tests
-    fi
+  # install stack and build project
+  - curl -sSL https://get.haskellstack.org/ | sh
+  - stack --version
+  - stack build --test --no-run-tests --bench --no-run-benchmarks
 
 script:
-  # HLint check
-  - curl -sSL https://raw.github.com/ndmitchell/neil/master/misc/travis.sh | sh -s -- hlint co-log-core/ co-log/src/ co-log-benchmark/
-  - |
-    if [ -z "$STACK_YAML" ]; then
-       cabal test all
-    else
-      stack test  co-log-core      --system-ghc
-      stack test  co-log           --system-ghc
-      stack test  co-log-polysemy  --system-ghc
-      stack build co-log-benchmark --system-ghc
-    fi
+  - stack test  co-log-core      --no-terminal
+  - stack test  co-log           --no-terminal
+  - stack test  co-log-polysemy  --no-terminal
+  - stack build co-log-benchmark --no-terminal
```

From the docs `--no-terminal` means `Enable/disable overriding terminal detection in the case of running in a false terminal`
I think we don't really need it :slightly_smiling_face: 
```suggestion
  - stack test  co-log-core
  - stack test  co-log
  - stack test  co-log-polysemy
  - stack build co-log-benchmark
```